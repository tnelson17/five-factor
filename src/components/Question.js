import React from 'react';
import styled from 'styled-components';

const RadioButton = styled.div`
  text-align: left;
  padding-left: 25%;
  &:hover {
    background-color: rgba(0, 0, 255, .1);
    border-radius: 10%;
    box-shadow: rgba(0, 0, 255, .1);
  }
`

class Question extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      clicked: false,
      value: '',
      domain: this.props.domain,
      id: this.props.id
    }
    this.handleChange = this.handleChange.bind(this)
  }

  handleChange(event) {
    let newValue = parseInt(event.target.value)

    this.props.handleQuestionSubmit({
      id: this.state.id,
      value: newValue,
      domain: this.state.domain
    })
    this.setState({
      clicked: true,
      value: newValue
    })
  }

  render(){
    return (
      <>
        <div> {this.props.text} </div>
          <RadioButton>
            <label>
              <input
                type="radio"
                value={4}
                checked={this.state.value === 4}
                onChange={this.handleChange}
              />
              Strongly Agree
            </label>
          </RadioButton>
          <RadioButton>
            <label>
              <input
                type="radio"
                value={3}
                checked={this.state.value === 3}
                onChange={this.handleChange}
              />
              Moderately Agree
            </label>
          </RadioButton>
          <RadioButton>
            <label>
              <input
                type="radio"
                value={2}
                checked={this.state.value === 2}
                onChange={this.handleChange}
              />
              Neither Agree nor Disagree
            </label>
          </RadioButton>
          <RadioButton>
            <label>
              <input
                type="radio"
                value={1}
                checked={this.state.value === 1}
                onChange={this.handleChange}
              />
              Moderately Disagree
            </label>
          </RadioButton>
          <RadioButton>
            <label>
              <input
                type="radio"
                value={0}
                checked={this.state.value === 0}
                onChange={this.handleChange}
              />
              Strongly Disagree
            </label>
          </RadioButton>
      </>
    )}
  }

export default Question;
