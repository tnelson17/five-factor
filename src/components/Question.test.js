import React from 'react';
import Question from './Question';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  shallow(<Question />)
})

it('renders correctly', () => {
  const tree = renderer
    .create(<Question />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
