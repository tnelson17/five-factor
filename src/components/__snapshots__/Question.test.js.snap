// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div>
     
     
  </div>,
  <div
    className="sc-bdVaJa hNRduX"
  >
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="radio"
        value={4}
      />
      Strongly Agree
    </label>
  </div>,
  <div
    className="sc-bdVaJa hNRduX"
  >
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="radio"
        value={3}
      />
      Moderately Agree
    </label>
  </div>,
  <div
    className="sc-bdVaJa hNRduX"
  >
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="radio"
        value={2}
      />
      Neither Agree nor Disagree
    </label>
  </div>,
  <div
    className="sc-bdVaJa hNRduX"
  >
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="radio"
        value={1}
      />
      Moderately Disagree
    </label>
  </div>,
  <div
    className="sc-bdVaJa hNRduX"
  >
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="radio"
        value={0}
      />
      Strongly Disagree
    </label>
  </div>,
]
`;
